docs/sources/auth/google.md

```diff
@@ -29,15 +29,18 @@ First, you need to create a Google OAuth Client:
 Specify the Client ID and Secret in the [Grafana configuration file]({{< relref "../administration/configuration.md#config-file-locations" >}}). For example:
 
 ```bash
+#################################### Google Auth #########################
 [auth.google]
-enabled = true
-client_id = CLIENT_ID
-client_secret = CLIENT_SECRET
+enabled = false
+allow_sign_up = true
+client_id = some_client_id
+client_secret =
 scopes = https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email
 auth_url = https://accounts.google.com/o/oauth2/auth
 token_url = https://accounts.google.com/o/oauth2/token
+api_url = https://www.googleapis.com/oauth2/v1/userinfo
 allowed_domains = mycompany.com mycompany.org
-allow_sign_up = true
+hosted_domain =
```

Comment:
From the [Google OAuth docs](https://developers.google.com/identity/protocols/oauth2/openid-connect):

> The hosted G Suite domain of the user. Provided only if the user belongs to a hosted domain.

> The hd (hosted domain) parameter streamlines the login process for G Suite hosted accounts. By including the domain of the G Suite user (for example, mycollege.edu), you can indicate that the account selection UI should be optimized for accounts at that domain. To optimize for G Suite accounts generally instead of just one domain, set a value of an asterisk (*): hd=*.

> Don't rely on this UI optimization to control who can access your app, as client-side requests can be modified. Be sure to validate that the returned ID token has an hd claim value that matches what you expect (e.g. mycolledge.edu). Unlike the request parameter, the ID token hd claim is contained within a security token from Google, so the value can be trusted.

So an optional parameter that makes the login flow a bit shorter by pre-selecting the G Suite domain and restricting logins to that domain e.g. would choose `grafana.com` if all the people logging in work at Grafana Labs. As noted above, not a security feature as it is possible to bypass it on the client side - more a convenience thing.

It is subtly different from the `allowed_domains` property which is a Grafana property and the logic for checking it is in the Grafana code base. `hosted_domain` is a Google OAuth property and is sent with the OAuth request to Google (so not part of the generic OAuth specific and only for Google OAuth).